internal/provider/gql.go

```diff
@@ -357,3 +357,86 @@ func (c *ExtendedGqlClient) SetScimGroupMapping(ctx context.Context, orgMrn stri
 
 	return c.Mutate(ctx, &setScimGroupMappingMutation, setScimGroupMappingInput, nil)
 }
+
+func (c *ExtendedGqlClient) UploadFramework(ctx context.Context, spaceMrn string, content []byte) error {
+	// Define the mutation struct according to the provided query
+	var uploadMutation struct {
+		UploadFramework bool `graphql:"uploadFramework(input: $input)"`
+	}
+
+	// Define the input variable according to the provided query
+	input := mondoov1.UploadFrameworkInput{
+		SpaceMrn: mondoov1.String(spaceMrn),
+		Dataurl:  mondoov1.String(newDataUrl(content)),
+	}
+
+	// Execute the mutation
+	return c.Mutate(ctx, &uploadMutation, input, nil)
+}
+
+type ComplianceFrameworkPayload struct {
+	Mrn      mondoov1.String
+	Name     mondoov1.String
+	State    mondoov1.String
+	ScopeMrn mondoov1.String
+}
+
+func (c *ExtendedGqlClient) GetFramework(ctx context.Context, spaceMrn string, spaceId string, uid string) (*ComplianceFrameworkPayload, error) {
+	// Define the query struct according to the provided query
+	var getFrameworkQuery struct {
+		ComplianceFramework ComplianceFrameworkPayload `graphql:"complianceFramework(input: $input)"`
+	}
+	frameworkMrn := fmt.Sprintf("//policy.api.mondoo.app/spaces/%s/frameworks/%s", spaceId, uid)
+	// Define the input variable according to the provided query
+	input := mondoov1.ComplianceFrameworkInput{
+		ScopeMrn:     mondoov1.String(spaceMrn),
+		FrameworkMrn: mondoov1.String(frameworkMrn),
+	}
+
+	variables := map[string]interface{}{
+		"input": input,
+	}
+
+	// Execute the query
+	err := c.Query(ctx, &getFrameworkQuery, variables)
+	if err != nil {
+		return nil, err
+	}
+
+	return &getFrameworkQuery.ComplianceFramework, nil
+}
+
+func (c *ExtendedGqlClient) UpdateFramework(ctx context.Context, frameworkMrn string, scopeMrn string, enabled bool) error {
+	var updateMutation struct {
+		ApplyFramework bool `graphql:"applyFrameworkMutation(input: $input)"`
+	}
+
+	if enabled {
```

Comment:
I would only set the action within the if statements and then to the mutate call once after the action was determined.